sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingAggregationSuite.scala

```diff
@@ -335,4 +344,67 @@ class StreamingAggregationSuite extends StreamTest with BeforeAndAfterAll {
       CheckLastBatch((90L, 1), (100L, 1), (105L, 1))
     )
   }
+
```

Comment:
I prefer to not add this complicated test because:
- TaskContext.get().addTaskCompletionListener has already been tested in other tests.
- Scala Reflection is not thread-safe in 2.10 and this test may be flaky since SQL also uses Scala Reflection.